File: ListViewDataBinding/app/src/main/java/com/example/vcfr67/androidtrainning/MainActivity.java
package com.example.vcfr67.androidtrainning;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.example.vcfr67.androidtrainning.databinding.ActivityMainBinding;
import com.example.vcfr67.androidtrainning.sample.SampleDataProvider;


public class MainActivity extends AppCompatActivity {
    private SampleDataProvider SampleDataProviderObj =  new SampleDataProvider();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(
                this, R.layout.activity_main);

        binding.setInfos(SampleDataProviderObj);
    }
}

File: architecture_component/part1/app/src/main/java/com/ressphere/architectureexample/MainActivity.kt
package com.ressphere.architectureexample

import android.os.Bundle
import android.os.StrictMode

import android.os.StrictMode.VmPolicy
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder()
                .detectDiskReads()
                .detectDiskWrites()
                .detectNetwork() // or .detectAll() for all detectable problems
                .penaltyLog()
                .build())
        StrictMode.setVmPolicy(VmPolicy.Builder()
                .detectLeakedSqlLiteObjects()
                .detectLeakedClosableObjects()
                .penaltyLog()
                .penaltyDeath()
                .build())
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val noteViewModel = ViewModelProvider(this).get(NoteViewModel::class.java)

        val noteAdapter = NoteAdapter()
        recycler_view.layoutManager = LinearLayoutManager(this)
        recycler_view.setHasFixedSize(true)
        recycler_view.adapter = noteAdapter
        noteViewModel.allNotes?.observe(this, Observer {
            noteAdapter.setNotes(it)
        })
    }
}